opentelekomcloud/data_source_opentelekomcloud_kms_data_key_v1.go

```diff
@@ -0,0 +1,70 @@
+package opentelekomcloud
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/huaweicloud/golangsdk/openstack/kms/v1/keys"
+	"time"
+)
+
+func dataSourceKmsDataKeyV1() *schema.Resource {
+	return &schema.Resource{
+		Read: resourceKmsDataKeyV1Read,
+
+		Schema: map[string]*schema.Schema{
+			"key_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"encryption_context": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"datakey_length": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"plain_text": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"cipher_text": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceKmsDataKeyV1Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	KmsDataKeyV1Client, err := config.kmsKeyV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack kms key client: %s", err)
+	}
+
+	req := &keys.DataEncryptOpts{
+		KeyID:             d.Get("key_id").(string),
+		EncryptionContext: d.Get("encryption_context").(string),
+		DatakeyLength:     d.Get("datakey_length").(string),
+	}
+	log.Printf("[DEBUG] KMS get data key for key: %s", d.Get("key_id").(string))
+	v, err := keys.DataEncryptGet(KmsDataKeyV1Client, req).ExtractDataKey()
+	if err != nil {
+		return err
+	}
+
+	d.SetId(time.Now().UTC().String())
```

Comment:
This can just be the `key_id` value.

Reply:
This is data key source, not key source. The data key source is issued by key source  If it set to key_id, it will be the same as key source.